dWorldServer/WorldServer.cpp

```diff
@@ -1291,14 +1291,17 @@ void WorldShutdownProcess(uint32_t zoneId) {
 }
 
 void WorldShutdownSequence() {
+	bool shouldShutdown = Game::ShouldShutdown() || worldShutdownSequenceComplete;
 	Game::lastSignal = -1;
 #ifndef DARKFLAME_PLATFORM_WIN32
-	if (Game::ShouldShutdown() || worldShutdownSequenceComplete)
+	if (shouldShutdown)
 #endif
 	{
 		return;
 	}
 
+	if (!Game::logger) return;
```

Comment:
Can we printf when this happens? This still method still feels weird in general.